File: snippets/nodejs-mongodb/express-student-api/models/Student.js
// models/Student.js — Equivalent to Student.java @Document model in Spring Boot
//
// Spring Boot:  @Document(collection = "students") public class Student { ... }
// Express:      We define the collection name and validation here

const { ObjectId } = require("mongodb");
const { getDB } = require("../db");

const COLLECTION = "students";

function getCollection() {
  return getDB().collection(COLLECTION);
}

// Validate student fields (equivalent to @Indexed, validation annotations)
function validate(student) {
  const errors = [];
  if (!student.name || student.name.trim() === "") errors.push("name is required");
  if (!student.rollNumber || student.rollNumber.trim() === "") errors.push("rollNumber is required");
  if (!student.department || student.department.trim() === "") errors.push("department is required");
  if (!student.email || student.email.trim() === "") errors.push("email is required");
  return errors;
}

// Create a clean student object (equivalent to the Student constructor)
function createStudentDoc(data) {
  return {
    name: data.name.trim(),
    rollNumber: data.rollNumber.trim(),
    department: data.department.trim(),
    email: data.email.trim(),
  };
}

module.exports = { getCollection, validate, createStudentDoc, ObjectId, COLLECTION };

File: labs/springboot-crud-mongodb/starter/src/main/java/com/lab/student/model/Student.java
package com.lab.student.model;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

@Document(collection = "students")
public class Student {
    @Id
    private String id;
    private String name;
    private String rollNumber;
    private String department;
    private String email;
    // TODO: Add a constructor with (name, rollNumber, department, email) and assign each field
    // TODO: Add getters and setters for all 5 fields (id, name, rollNumber, department, email)
}
